checkov/kustomize/runner.py

```diff
@@ -413,8 +509,23 @@ def _get_kubectl_output(filePath: str, template_renderer_command: str, source_ty
             template_render_command_options = "kustomize"
         if template_renderer_command == "kustomize":
             template_render_command_options = "build"
-        proc = subprocess.Popen([template_renderer_command, template_render_command_options, filePath], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
+
+        add_origin_annotations_return_code = None
+
+        if env_vars_config.ALLOW_KUSTOMIZE_FILE_EDITS:
+            add_origin_annotations_command = 'kustomize edit add buildmetadata originAnnotations'
+            add_origin_annotations_return_code = subprocess.Popen(add_origin_annotations_command.split(' '),  # nosec
+                                                                  cwd=filePath).wait()
+
+        full_command = f'{template_renderer_command} {template_render_command_options}'
+        proc = subprocess.Popen(full_command.split(' '), cwd=filePath, stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
         output, _ = proc.communicate()
+
+        if env_vars_config.ALLOW_KUSTOMIZE_FILE_EDITS and add_origin_annotations_return_code == 0:
+            # If the return code is not 0, we didn't add the new buildmetadata field, so we shouldn't remove it
+            remove_origin_annotaions = 'kustomize edit remove buildmetadata originAnnotations'
+            subprocess.Popen(remove_origin_annotaions.split(' '), cwd=filePath).wait()  # nosec
```

Comment:
not sure, why we use `Popen` better to use `run` instead as recommend by Python, at least for the second call we can switch to `run`

Reply:
tbh I just used popen as that's what the function already used. will do it now